src/format.rs

```diff
@@ -72,14 +72,14 @@ pub struct UrlNoFragmentFormatter<'a> {
 
 impl<'a> Show for UrlNoFragmentFormatter<'a> {
     fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        try!(formatter.write(self.url.scheme.as_bytes()));
-        try!(formatter.write(b":"));
+        try!(formatter.write_str(&*self.url.scheme));
+        try!(formatter.write_str(":"));
         try!(self.url.scheme_data.fmt(formatter));
         match self.url.query {
             None => (),
             Some(ref query) => {
-                try!(formatter.write(b"?"));
-                try!(formatter.write(query.as_bytes()));
+                try!(formatter.write_str("?"));
+                try!(formatter.write_str(&**query));
```

Comment:
Is `&**query` preferable to `query.as_slice()`? (Same for `&*self.url.scheme` v.s. `self.url.scheme.as_slice()`)